### PR TITLE
User

### PR DESCRIPTION
**Pull request by sidewinder2020**
LIST OF CHANGES:
-controllers added: Users, Sessions, home
- models added: user
- migration added: user
- views added: user, sessions, home

THINGS I'M UNSURE ON:
- redirects - need to fit user story
- how to combine with other features
-

RSPEC GREEN?
YES
NO- WHY?

*MENTION GROUP MEMBER TO REVIEW PR*

### CODE REVIEW

**Comment by blsrofe** [comment]
This is awesome!

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class HomeController < ApplicationController
```
Comment (snayrouz) [inline app/controllers/home_controller.rb:1]: thanks for setting up the home controller

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(username: params[:session][:username])
+    if user && user.authenticate(params[:session][:username])
+      session[:user_id] = user.id
+      session[:logged_in?] = true
```
Comment (snayrouz) [inline app/controllers/sessions_controller.rb:10]: like the logic here

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:first_name, :last_name, :username, :email, :password, :address)
+  end
+
+  def require_login
+    unless session[:logged_in?]
```
Comment (snayrouz) [inline app/controllers/users_controller.rb:30]: nice private method

**Review by ButeraMV — COMMENTED** [review]
Resolved merge conflicts.

File: app/views/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<%= form_for(:session, url: login_path) do |f| %>
+  <%= f.label :username %>
+  <%= f.text_field :username %>
+  <%= f.label :password %>
+  <%= f.password_field :password %>
+  <%= f.submit "Login" %>
```
Comment (snayrouz) [inline app/views/sessions/new.html.erb:6]: great form, love the execution

**Pull request merged**